src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractEditProjectFileCommand.cs

```diff
@@ -0,0 +1,132 @@
+        private readonly IFileSystem _fileSystem;
+        private readonly ITextDocumentFactoryService _textDocumentFactoryService;
+        private readonly IVsEditorAdaptersFactoryService _editorFactoryService;
+        private readonly IProjectThreadingService _threadingService;
+        private readonly IVsShellUtilitiesHelper _shellUtilities;
+
+        public AbstractEditProjectFileCommand(UnconfiguredProject unconfiguredProject,
+            IProjectCapabilitiesService projectCapabilitiesService,
+            IServiceProvider serviceProvider,
+            IMsBuildAccessor msbuildAccessor,
+            IFileSystem fileSystem,
+            ITextDocumentFactoryService textDocumentService,
+            IVsEditorAdaptersFactoryService editorFactoryService,
+            IProjectThreadingService threadingService,
+            IVsShellUtilitiesHelper shellUtilities)
+        {
+            _unconfiguredProject = unconfiguredProject;
+            _projectCapabiltiesService = projectCapabilitiesService;
+            _serviceProvider = serviceProvider;
+            _msbuildAccessor = msbuildAccessor;
+            _fileSystem = fileSystem;
+            _textDocumentFactoryService = textDocumentService;
+            _editorFactoryService = editorFactoryService;
+            _threadingService = threadingService;
+            _shellUtilities = shellUtilities;
+        }
+
+        protected override Task<CommandStatusResult> GetCommandStatusAsync(IProjectTree node, bool focused, string commandText, CommandStatus progressiveStatus) =>
+            ShouldHandle(node) ?
+                GetCommandStatusResult.Handled(GetCommandText(node), CommandStatus.Enabled) :
+                GetCommandStatusResult.Unhandled;
+
+        protected override async Task<bool> TryHandleCommandAsync(IProjectTree node, bool focused, Int64 commandExecuteOptions, IntPtr variantArgIn, IntPtr variantArgOut)
+        {
+            if (!ShouldHandle(node)) return false;
+
+            var caption = Path.GetFileName(_unconfiguredProject.FullPath);
+
+            var projPath = await GetFile().ConfigureAwait(false);
+            await _threadingService.SwitchToUIThread();
+            IVsWindowFrame frame;
+
+            frame = _shellUtilities.OpenDocumentWithSpecificEditor(_serviceProvider, projPath, XmlEditorFactoryGuid, Guid.Empty);
+
+            // Clean up the caption of the window
+            Verify.HResult(frame.SetProperty((int)__VSFPROPID5.VSFPROPID_OverrideCaption, caption));
+
+            // When the document is closed, clean up the file on disk
+            var fileCleanupListener = new EditProjectFileCleanupFrameNotifyListener(projPath, _fileSystem);
+            Verify.HResult(frame.SetProperty((int)__VSFPROPID.VSFPROPID_ViewHelper, fileCleanupListener));
+
+            // Ensure that the window is not reopened when the solution is closed
+            Verify.HResult(frame.SetProperty((int)__VSFPROPID5.VSFPROPID_DontAutoOpen, true));
```

Comment:
If the vs is closed with the window open, would it still prevent the editor from opening when VS is opened?